frontend/src/features/public-form/components/FormFields/PublicFormSubmitButton.tsx

```diff
@@ -57,7 +68,17 @@ export const PublicFormSubmitButton = ({
 
   const checkBeforeOpen = async () => {
     const result = await trigger()
-    if (result) onOpen()
+
+    if (result) {
+      // get previous payment
+      const payment = await getPreviousPayment(paymentEmailField.value, formId)
```

Comment:
Just to verify: is `paymentEmailField.value` guaranteed to be normalized? (trimmed + lowercased?)

Reply:
It is guaranteed! On the FE, we will lowercase any inputs into the email field -> so the formInputs email field is guaranteed to be normalized! thanks @LinHuiqing 🙏